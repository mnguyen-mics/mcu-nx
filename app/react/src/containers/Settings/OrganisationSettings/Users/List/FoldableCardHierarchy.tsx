import { Timeline } from 'antd';
import { Content } from 'antd/lib/layout/layout';
import * as React from 'react';
import FoldableCard, { FoldableCardProps } from './FoldableCard';

export interface FoldableCardHierarchyResource {
  foldableCard: FoldableCardProps;
  children?: FoldableCardHierarchyResource[];
}

export interface FoldableCardHierarchyProps {
  hierachy?: FoldableCardHierarchyResource;
  className?: string;
  cardClassName?: string;
}

export default class FoldableCardHierarchy extends React.Component<FoldableCardHierarchyProps> {
  buildFoldableCardHierachy() {
    const recusiveBuildFoldableCardHierachy = (
      currentNode: FoldableCardHierarchyResource,
    ): React.ReactNode => {
      const childrenNodes = [];

      if (currentNode.children) {
        for (let index = 0; index < currentNode.children.length; index++) {
          childrenNodes.push(
            <Timeline.Item
              key={`mcs-foldable-card-timelineItem-${index}`}
              className={
                'mcs-foldable-card-timeline' +
                (index === currentNode.children.length - 1
                  ? ' mcs-foldable-card-timeline-last'
                  : '')
              }
              dot={
                <div className='mcs-foldable-card-timeline-dash-container'>
                  <hr className='mcs-foldable-card-timeline-dash' />
                </div>
              }
            >
              <div className='mcs-foldable-card-children-container'>
                {recusiveBuildFoldableCardHierachy(currentNode.children[index])}
              </div>
            </Timeline.Item>,
          );
        }
      }

      return (
        <div className='mcs-foldableCardHierachy_container'>
          <FoldableCard {...currentNode.foldableCard} />
          <div className='mcs-foldable-card-children-container-first'>{childrenNodes}</div>
        </div>
      );
    };

    const { hierachy } = this.props;

    return hierachy ? recusiveBuildFoldableCardHierachy(hierachy) : <div />;
  }

  render() {
    return <Content>{this.buildFoldableCardHierachy()}</Content>;
  }
}
