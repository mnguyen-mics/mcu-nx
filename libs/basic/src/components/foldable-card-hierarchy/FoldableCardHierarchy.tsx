import { Timeline } from 'antd';
import { Content } from 'antd/lib/layout/layout';
import React from 'react';
import FoldableCard, { FoldableCardProps } from './FoldableCard';

export interface FoldableCardHierarchyResource {
  foldableCard: FoldableCardProps;
  children: FoldableCardHierarchyResource[];
}

export interface FoldableCardHierarchyProps {
  hierachy?: FoldableCardHierarchyResource;
  className?: string;
  cardClassName?: string;
}

export default class FoldableCardHierarchy extends React.Component<FoldableCardHierarchyProps> {
  buildFoldableCardHierachy() {
    const recusiveBuildFoldableCardHierachy = (
      currentNode: FoldableCardHierarchyResource,
    ): React.ReactNode => {
      const childrenNodes = [];

      for (let index = 0; index < currentNode.children.length; index++) {
        childrenNodes.push(
          <Timeline.Item
            className={
              'mcs-foldable-card-timeline' +
              (index === currentNode.children.length - 1 ? ' mcs-foldable-card-timeline-last' : '')
            }
            dot={
              <div className='mcs-foldable-card-timeline-dash-container'>
                <hr className='mcs-foldable-card-timeline-dash' />
              </div>
            }
          >
            <div className='mcs-foldable-card-children-container'>
              {recusiveBuildFoldableCardHierachy(currentNode.children[index])}
            </div>
          </Timeline.Item>,
        );
      }

      return (
        <React.Fragment>
          <div>
            <FoldableCard {...currentNode.foldableCard} />
            <div className='mcs-foldable-card-children-container-first'>{childrenNodes}</div>
          </div>
        </React.Fragment>
      );
    };

    const { hierachy } = this.props;

    return hierachy ? recusiveBuildFoldableCardHierachy(hierachy) : <div />;
  }

  render() {
    return <Content style={{ padding: '15px' }}>{this.buildFoldableCardHierachy()}</Content>;
  }
}
