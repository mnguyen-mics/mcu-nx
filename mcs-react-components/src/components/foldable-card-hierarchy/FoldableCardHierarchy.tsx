import { Timeline } from 'antd';
import { Content } from 'antd/lib/layout/layout';
import * as React from 'react';
import FoldableCard, { FoldableCardProps } from './FoldableCard';

export interface FoldableCardHierarchyResource {
  foldableCard: FoldableCardProps;
  children: FoldableCardHierarchyResource[];
}

export interface FoldableCardHierarchyProps {
  hierachy?: FoldableCardHierarchyResource;
  className?: string;
  cardClassName?: string;
}

export default class FoldableCardHierarchy extends React.Component<FoldableCardHierarchyProps> {
  buildFoldableCardHierachy() {
    const recusiveBuildFoldableCardHierachy = (
      currentNode: FoldableCardHierarchyResource,
    ): React.ReactNode => {
      const childrenNodes = [];

      for (let index = 0; index < currentNode.children.length; index++) {
        childrenNodes.push(
          <Timeline.Item
            className={
              'mics-foldable-card-timeline' +
              (index === currentNode.children.length - 1 ? ' mics-foldable-card-timeline-last' : '')
            }
            dot={
              <div className='mics-foldable-card-timeline-dash-container'>
                <hr className='mics-foldable-card-timeline-dash' />
              </div>
            }
          >
            <div className='mics-foldable-card-children-container'>
              {recusiveBuildFoldableCardHierachy(currentNode.children[index])}
            </div>
          </Timeline.Item>,
        );
      }

      return (
        <React.Fragment>
          <div>
            <FoldableCard {...currentNode.foldableCard} />
            <div className='mics-foldable-card-children-container-first'>{childrenNodes}</div>
          </div>
        </React.Fragment>
      );
    };

    const { hierachy } = this.props;

    return hierachy ? recusiveBuildFoldableCardHierachy(hierachy) : <div />;
  }

  render() {
    return <Content style={{ padding: '15px' }}>{this.buildFoldableCardHierachy()}</Content>;
  }
}
