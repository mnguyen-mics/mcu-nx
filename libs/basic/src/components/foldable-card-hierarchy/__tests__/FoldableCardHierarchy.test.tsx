import 'jest';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import FoldableCardHierarchy, { FoldableCardHierarchyProps } from '../FoldableCardHierarchy';

it('renders the FoldableCardHierarchy', () => {
  const props: FoldableCardHierarchyProps = {
    hierachy: {
      foldableCard: {
        isDefaultActive: false,
        header: 'Org1',
        body: 'someOrg1Users',
      },
      children: [
        {
          foldableCard: {
            isDefaultActive: true,
            header: 'Org1.1',
            body: (
              <div>
                <div>someOrg11Users</div>
                <div>someOrg11Users</div>
              </div>
            ),
          },
          children: [
            {
              foldableCard: {
                isDefaultActive: false,
                header: 'Org1.1.1',
                body: 'someOrg111Users',
              },
              children: [],
            },
          ],
        },
        {
          foldableCard: {
            isDefaultActive: false,
            header: 'Org1.2',
            body: 'someOrg12Users',
          },
          children: [],
        },
      ],
    },
  };
  const component = TestRenderer.create(<FoldableCardHierarchy {...props} />);
  const res = component.toJSON();
  expect(res).toMatchSnapshot();
});
