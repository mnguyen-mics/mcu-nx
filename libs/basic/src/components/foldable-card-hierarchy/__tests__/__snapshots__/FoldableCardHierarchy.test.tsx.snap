// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the FoldableCardHierarchy 1`] = `
<main
  className="ant-layout-content"
  style={
    Object {
      "padding": "15px",
    }
  }
>
  <div>
    <main
      className="ant-layout-content"
    >
      <div
        className="ant-collapse ant-collapse-icon-position-start ant-collapse-ghost mcs-foldable-card "
        role={null}
      >
        <div
          className="ant-collapse-item mcs-foldable-card"
        >
          <div
            aria-disabled={false}
            aria-expanded={false}
            className="ant-collapse-header"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="ant-collapse-expand-icon"
              onClick={null}
            >
              <span
                aria-label="plus-circle"
                className="anticon anticon-plus-circle ant-collapse-arrow"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="plus-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M696 480H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
                  />
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                  />
                </svg>
              </span>
            </div>
            <span
              className="ant-collapse-header-text"
              onClick={null}
            >
              Org1
            </span>
          </div>
        </div>
      </div>
    </main>
    <div
      className="mcs-foldable-card-children-container-first"
    >
      <li
        className="ant-timeline-item mcs-foldable-card-timeline"
      >
        <div
          className="ant-timeline-item-tail"
        />
        <div
          className="ant-timeline-item-head ant-timeline-item-head-custom ant-timeline-item-head-blue"
          style={
            Object {
              "borderColor": undefined,
              "color": undefined,
            }
          }
        >
          <div
            className="mcs-foldable-card-timeline-dash-container"
          >
            <hr
              className="mcs-foldable-card-timeline-dash"
            />
          </div>
        </div>
        <div
          className="ant-timeline-item-content"
        >
          <div
            className="mcs-foldable-card-children-container"
          >
            <div>
              <main
                className="ant-layout-content"
              >
                <div
                  className="ant-collapse ant-collapse-icon-position-start ant-collapse-ghost mcs-foldable-card "
                  role={null}
                >
                  <div
                    className="ant-collapse-item ant-collapse-item-active mcs-foldable-card"
                  >
                    <div
                      aria-disabled={false}
                      aria-expanded={true}
                      className="ant-collapse-header"
                      onClick={[Function]}
                      onKeyPress={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <div
                        className="ant-collapse-expand-icon"
                        onClick={null}
                      >
                        <span
                          aria-label="minus-circle"
                          className="anticon anticon-minus-circle ant-collapse-arrow"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="minus-circle"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M696 480H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h368c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
                            />
                            <path
                              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                            />
                          </svg>
                        </span>
                      </div>
                      <span
                        className="ant-collapse-header-text"
                        onClick={null}
                      >
                        Org1.1
                      </span>
                    </div>
                    <div
                      className="ant-collapse-content ant-collapse-content-active"
                      role={null}
                    >
                      <div
                        className="ant-collapse-content-box"
                      >
                        <div>
                          <div>
                            someOrg11Users
                          </div>
                          <div>
                            someOrg11Users
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </main>
              <div
                className="mcs-foldable-card-children-container-first"
              >
                <li
                  className="ant-timeline-item mcs-foldable-card-timeline mcs-foldable-card-timeline-last"
                >
                  <div
                    className="ant-timeline-item-tail"
                  />
                  <div
                    className="ant-timeline-item-head ant-timeline-item-head-custom ant-timeline-item-head-blue"
                    style={
                      Object {
                        "borderColor": undefined,
                        "color": undefined,
                      }
                    }
                  >
                    <div
                      className="mcs-foldable-card-timeline-dash-container"
                    >
                      <hr
                        className="mcs-foldable-card-timeline-dash"
                      />
                    </div>
                  </div>
                  <div
                    className="ant-timeline-item-content"
                  >
                    <div
                      className="mcs-foldable-card-children-container"
                    >
                      <div>
                        <main
                          className="ant-layout-content"
                        >
                          <div
                            className="ant-collapse ant-collapse-icon-position-start ant-collapse-ghost mcs-foldable-card "
                            role={null}
                          >
                            <div
                              className="ant-collapse-item mcs-foldable-card"
                            >
                              <div
                                aria-disabled={false}
                                aria-expanded={false}
                                className="ant-collapse-header"
                                onClick={[Function]}
                                onKeyPress={[Function]}
                                role="button"
                                tabIndex={0}
                              >
                                <div
                                  className="ant-collapse-expand-icon"
                                  onClick={null}
                                >
                                  <span
                                    aria-label="plus-circle"
                                    className="anticon anticon-plus-circle ant-collapse-arrow"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="plus-circle"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="64 64 896 896"
                                      width="1em"
                                    >
                                      <path
                                        d="M696 480H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
                                      />
                                      <path
                                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                                      />
                                    </svg>
                                  </span>
                                </div>
                                <span
                                  className="ant-collapse-header-text"
                                  onClick={null}
                                >
                                  Org1.1.1
                                </span>
                              </div>
                            </div>
                          </div>
                        </main>
                        <div
                          className="mcs-foldable-card-children-container-first"
                        />
                      </div>
                    </div>
                  </div>
                </li>
              </div>
            </div>
          </div>
        </div>
      </li>
      <li
        className="ant-timeline-item mcs-foldable-card-timeline mcs-foldable-card-timeline-last"
      >
        <div
          className="ant-timeline-item-tail"
        />
        <div
          className="ant-timeline-item-head ant-timeline-item-head-custom ant-timeline-item-head-blue"
          style={
            Object {
              "borderColor": undefined,
              "color": undefined,
            }
          }
        >
          <div
            className="mcs-foldable-card-timeline-dash-container"
          >
            <hr
              className="mcs-foldable-card-timeline-dash"
            />
          </div>
        </div>
        <div
          className="ant-timeline-item-content"
        >
          <div
            className="mcs-foldable-card-children-container"
          >
            <div>
              <main
                className="ant-layout-content"
              >
                <div
                  className="ant-collapse ant-collapse-icon-position-start ant-collapse-ghost mcs-foldable-card "
                  role={null}
                >
                  <div
                    className="ant-collapse-item mcs-foldable-card"
                  >
                    <div
                      aria-disabled={false}
                      aria-expanded={false}
                      className="ant-collapse-header"
                      onClick={[Function]}
                      onKeyPress={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <div
                        className="ant-collapse-expand-icon"
                        onClick={null}
                      >
                        <span
                          aria-label="plus-circle"
                          className="anticon anticon-plus-circle ant-collapse-arrow"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="plus-circle"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M696 480H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
                            />
                            <path
                              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                            />
                          </svg>
                        </span>
                      </div>
                      <span
                        className="ant-collapse-header-text"
                        onClick={null}
                      >
                        Org1.2
                      </span>
                    </div>
                  </div>
                </div>
              </main>
              <div
                className="mcs-foldable-card-children-container-first"
              />
            </div>
          </div>
        </div>
      </li>
    </div>
  </div>
</main>
`;
